spec/system/admin/feature_flags_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Feature Flags' do
+  context 'when an admin user' do
+    before do
+      user = create(:user, admin: true)
+      sign_in(user)
+    end
+
+    it 'allows admins to enable and disable features' do
+      visit home_path
+      expect(page).not_to have_content('test feature is enabled')
+
+      visit admin_dashboard_path
+      find_by_id('utility_nav').click
+      click_on 'Feature Flags'
+      click_on 'Add Feature'
+      fill_in 'value', with: 'test_feature'
+      click_on 'Add Feature'
+      click_on 'Fully Enable'
+
+      visit home_path
+      expect(page).to have_content('test feature is enabled')
+    end
+  end
+
+  context 'when a learner user' do
+    before do
+      user = create(:user, admin: false)
+      sign_in(user)
+    end
+
+    it 'does not allow learners to enable and disable features' do
+      visit admin_dashboard_path
+
+      expect(page).not_to have_content('Unauthorized access')
```

Comment:
Is this expectation the wrong way around or have I misunderstood? What is the expected behaviour, a redirect?

Reply:
Great spot, yep its the wrong way around. The ideal thing to test here would be this flash appearing after trying to access the feature flags url directly. But those routes don't exist unless you're signed in as an admin user. Capybara just raises a routing error when you try to access it. This is the next best thing to have an expectation for.